sklearn/manifold/t_sne.py

```diff
@@ -521,5 +836,54 @@ def fit_transform(self, X):
         X_new : array, shape (n_samples, n_components)
             Embedding of the training data in low-dimensional space.
         """
-        self._fit(X)
+        embedding = self._fit(X)
+        self.embedding_ = embedding
         return self.embedding_
+
+    def fit(self, X):
+        """Fit X into an embedded space.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features) or (n_samples, n_samples)
+            If the metric is 'precomputed' X must be a square distance
+            matrix. Otherwise it contains a sample per row. If the method
+            is 'standard', X may be a sparse matrix of type 'csr', 'csc'
+            or 'coo'.
+        """
+        self.fit_transform(X)
+
+    def transform(self, X):
+        """Transform X to a previously fit embedded space.
+
+        A previous training set must have already been fit.
+        The new gradient is calculated using contributions from
+        previously fit data, but only the new data is transformed.
+        This is not the equivalent of running fit_transform,
+        since calling fit(X) followed by transform(X) runs the
```

Comment:
I'm not a huge fan of this behavior either, but agreed, there is no alternative. I think the best we can do is issue a warning?